multiqc/modules/hicpro/hicpro.py

```diff
@@ -0,0 +1,386 @@
+            pairs involving multi-mapped reads are usually discarded. Note that the filtering 
+            at pairing level can change accrding to the parameters used.''',
+            plot = self.hicpro_pairing_chart()
+        )
+        
+        self.add_section (
+            name = 'Read Pair Filtering',
+            anchor = 'hicpro-filtering',
+            description = 'Filtering of read pairs.',
+            helptext = '''
+            Aligned read pairs are filtered to select valid 3C products from two 
+            different restriction fragments. Read pairs coming from the same fragments, 
+            such as self-ligation or unligated (danging-end) fragments, are discarded.
+            Ligation products involving neighboring restriction fragment (religation) are also discarded.
+            Finaly, as the ligation should be a random process, valid read pairs from all 
+            orientations (R=Reverse, F=forward) are expected to be observed at the same proportion.''',
+            plot = self.hicpro_filtering_chart()
+        )
+
+        self.add_section (
+            name = 'Contact Statistics',
+            anchor = 'hicpro-rmdup',
+            description = 'Contacts statistics after duplicates removal.',
+            helptext = '''
+            Description of contact frequency after duplicates removal. Intra-chromosomal 
+            (*cis*) interaction are expected to be more frequent than inter-chromosomal contacts (*trans*)''',
+            plot = self.hicpro_contact_chart()
+        )
+
+        allele_plot = self.hicpro_as_chart()
+        if allele_plot:
+            self.add_section (
+                name = 'Allele-specific analysis',
+                anchor = 'hicpro-asan',
+                description = 'Assignment of valid interactions (afer duplicates removal) to parental alleles.',
+                helptext = '''
+                Description of allele-specific contacts. Valid interactions (0-1 and 1-1, resp. 0-2, 2-2) 
+                are used to build the genome1 (resp. genome2) parental maps.''',
+                plot = self.hicpro_as_chart()
+            )
+            
+
+
+    def parse_hicpro_stats(self, f):
+        """ Parse a HiC-Pro stat file """
+        s_name = self.clean_s_name(os.path.basename(f['root']), os.path.dirname(f['root']))
+
+        if s_name not in self.hicpro_data.keys():
+            self.hicpro_data[s_name] = {}
+
+        self.add_data_source(f, s_name)
+        for l in f['f'].splitlines():
+            if not l.startswith('#'):
+                s = l.split("\t")
+                if s[0] in self.hicpro_data[s_name]:
+                    log.debug("Duplicated keys found! Overwriting: {}".format(s[0]))
+                self.hicpro_data[s_name][s[0]] = int(s[1])
+
+    def hicpro_stats_table(self):
+         """ Add HiC-Pro stats to the general stats table """
+         headers = OrderedDict()
+         headers['total_R1'] = {
```

Comment:
Minor thing: General Stats table columns typically follow the chronological order of processing, with the earliest steps on the right hand side of the table and the later steps on the left. So it will make more sense if this is the final column defined in this function (so that it ends up on the right of the table).

Reply:
Really ! you usually put the first step on the right (so total number of reads at the end ?) I would do the reverse no ? first step on the left ? and then from the left to the right, you see how many reads you are loosing at each step of the pipeline ...
But tell me, in this case, I have to reorder the columns

Reply:
hah, for long and historical reasons, yes that's the way it is. See for example this bisulfite report: http://multiqc.info/examples/bs-seq/multiqc_report.html

Reply:
ok. I will change that (and call Felix to tell him that the bismark report is not in the right order :smile: !)

Reply:
Hah, it’s not Felix’s fault (for once!), I wrote the Bismark MultiQC code. It’s because the General Stats table often gets very wide and scrolls out of view. The columns on the left are always immediately visible, so I thought that it makes sense to have the final conclusion columns there. Then you can scroll back in time, so to speak.

But yes, it can be a little counter intuitive. Right to left reading and bottom to top category ordering.

Reply:
I'm glad to hear that this is not my fault!

Keep up the good work guys (and spamming my inbox with emails every few minutes)!!

Reply:
:+1: Sorry for the spam !!!